### PR TITLE
feat: subscribe to js event listener

### PR DESCRIPTION
**Pull request by Brian-Ding**
[[中文版模板 / Chinese template](https://github.com/ant-design-blazor/ant-design-blazor/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

Subscribe to the js event listener instead of calling js multiple times
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/core/JsInterop/DomEventService.cs [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
     public class DomEventService
     {
-        private Dictionary<string, Func<object>> _domEventListeners = new Dictionary<string, Func<object>>();
+        private object _subscriptionsLock = new object();
+        private Dictionary<string, List<Subscription>> _domEventSubscriptions = new Dictionary<string, List<Subscription>>();
+        private Dictionary<string, List<Subscription>> DomEventSubscriptions
+        {
+            get
+            {
+                lock (_subscriptionsLock)
```
Comment (diegofrata) [inline components/core/JsInterop/DomEventService.cs:17]: What's the purpose of this lock? You're locking over the getter and this won't safeguard any operations on the dictionary.

Comment (ElderJames) [reply components/core/JsInterop/DomEventService.cs:17]: How about use `CconcurrentDictionary`?

**Comment by Brian-Ding** [comment]
1. replace dictionary with concurrent dictionary
2. add an optional flag to share js listener or not.

**Review by ElderJames — APPROVED** [review]
LGTM.

**Pull request merged**